lib/asset_sync/storage.rb

```diff
@@ -264,7 +264,7 @@ def upload_file(f)
       if config.compression
         compressed_name = "#{path}/#{f}.#{config.compression}"
 
-        if File.extname(f) == config.compression
+        if File.extname(f)[1..-1] == config.compression
```

Comment:
Just comment, I will commit the change myself
```suggestion
        # `File.extname` returns value with `.` prefix, `config.compression` contains value without `.`
        if File.extname(f)[1..-1] == config.compression
```